plugins/outputs/sql/sql.go

```diff
@@ -98,7 +98,14 @@ func (p *SQL) Init() error {
 }
 
 func (p *SQL) Connect() error {
-	db, err := gosql.Open(p.Driver, p.DataSourceName)
+	secretBuffer, err := p.DataSourceName.Get()
+	if err != nil {
+		return fmt.Errorf("loading data source name secret failed: %w", err)
+	}
+	dsn := secretBuffer.String()
+	secretBuffer.Destroy()
+
+	db, err := gosql.Open(p.Driver, dsn)
```

Comment:
```suggestion
	defer secretBuffer.Destroy()

	db, err := gosql.Open(p.Driver, secretBuffer.String())
```
Maybe also rename `secretBuffer` to `dataSourceName`?

Reply:
I was told a while ago that secrets should be destroyed as soon as possible, that's why I didn't put it in defer. But I can change it if needed.

Reply:
Okay, I see that's indeed used when the method returns just after. So what about this?
```suggestion
	db, err := gosql.Open(p.Driver, secretBuffer.String())
	secretBuffer.Destroy()
```

Reply:
> Would it make sense to set the value to `""` in `func (p *SQL) Connect()` after Open is done?

What does that gain above my suggested solution?

Reply:
Okay, tell me what to do and I'll do it :D 
It doesn't really matter to me, if what you say is true and Open does not wait on a connection to a DB, then this is fine.